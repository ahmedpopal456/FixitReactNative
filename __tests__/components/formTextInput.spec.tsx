import React from 'react';
import 'react-native';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import FormTextInput from '../../src/components/formTextInput';

describe('Form Text Input', () => {
  const onChange = jest.fn();
  const onFocus = jest.fn();
  const onBlur = jest.fn();
  const props = {
    onChange,
    onFocus,
    onBlur,
    big: false,
    value: 'some value',
    title: 'some title',
    padLeft: false,
    numeric: false,
  };

  it('renders correctly', async () => {
    const formTextInput = renderer.create(
      <FormTextInput {...props} />,
    ).toJSON();
    expect(formTextInput).toMatchSnapshot();
  });

  it('called onFocus', async () => {
    const wrapper = shallow(
      <FormTextInput {...props} />,
    );
    wrapper.instance().onFocus();
    expect(onFocus).toHaveBeenCalledTimes(1);
  });

  it('called onBlur', async () => {
    const wrapper = shallow(
      <FormTextInput {...props} />,
    );
    wrapper.instance().onBlur();
    expect(onBlur).toHaveBeenCalledTimes(1);
  });
});
