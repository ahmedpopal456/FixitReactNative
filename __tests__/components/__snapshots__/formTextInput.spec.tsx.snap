// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Text Input renders correctly 1`] = `
<View
  style={
    Object {
      "position": "relative",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "elevation": 1,
        "left": 10,
        "position": "absolute",
        "top": -10,
        "zIndex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "backgroundColor": "#fff",
          "fontSize": 13,
          "padding": 3,
        }
      }
    >
      some title
    </Text>
  </View>
  <TextInput
    allowFontScaling={true}
    keyboardType="default"
    multiline={false}
    onBlur={[Function]}
    onChangeText={[MockFunction]}
    onFocus={[Function]}
    rejectResponderTermination={true}
    selectionColor="#FFD14A"
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#8B8B8B",
          "borderRadius": 8,
          "borderWidth": 1,
          "paddingLeft": 10,
          "paddingRight": 10,
        },
        Object {
          "height": 50,
          "paddingLeft": 10,
        },
      ]
    }
    underlineColorAndroid="transparent"
    value="some value"
  />
</View>
`;
